### DISCUSSION TITLE
Adding prepareForValidation method

### DISCUSSION THREAD

**Discussion by robert-abram** [spatie/laravel-data#324] [Q&A]
```prepareForValidation``` change only data before validation and data are intact but ```prepareForPipeline``` change data. Is there any option to prepare date before validation, similar to prepareForValidation https://laravel.com/docs/9.x/validation#preparing-input-for-validation in FormRequest ?

**Comment by Nir-An** [comment]
Hi @abrampl  :+1: 

as I understand it should work the same as ``prepareForValidation``
just change the data and return it
 ```php

    public static function prepareForPipeline(Collection $properties) : Collection
    {
        $properties->put('metadata', $properties->only(['release_year', 'producer']));
        
        return $properties;
    }
```
[original discussion](https://github.com/spatie/laravel-data/discussions/14)
[Docs: preparing-data-for-the-pipeline](https://spatie.be/docs/laravel-data/v2/advanced-usage/pipeline#content-preparing-data-for-the-pipeline)

**Reply by robert-abram** [reply]
Nope it's not working the same @Nir-An  , when you use prepareForPipeline, it will change you data, for example you create dto to check characters count, so before you check max size of string you want to strip_tags before you check size, if you do 

```php
 public static function prepareForPipeline(Collection $properties) : Collection
   {
       $properties->put('description', strip_tags($properties->only(['rdescription'])));
       
       return $properties;
   }

```

then of course before validation string is without tags, but after validation you want to save string with tags to db, and if you check description after dto is create and validated, it's show that value was overwriten, so in fact you loose orginal value of descrription
